src/main/java/EventsAndCommands/ChatEventHandler.java

```diff
@@ -0,0 +1,47 @@
+package EventsAndCommands;
+
+
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+
+
+/**
+ * This class provides the base structure of a chatevent that might be handled. 
+ * To implement a new ChatEventHandler it is sufficient to just inherit from this 
+ * Class and override {@see ChatEventHandler#trigger(GuildMessageReceivedEvent)}
+ * and {@see ChatEventHandler#receiveCommand(GuildMessageReceivedEvent)}.
+ */
+public abstract class ChatEventHandler {
+
+    public ChatEventHandler(){
+        register();
```

Comment:
Having this kind of side-effect in a constructor sounds really dangerous. I'd prefer to make the (un) register methods public and let the auto-discovery handle it.

Maybe adjust the discovery to:
```java
loadAllClasses(Class<T> superType, Consumer<T> initialization)
```